std/algorithm/searching.d

```diff
@@ -1063,6 +1063,29 @@ if (isBidirectionalRange!R1 &&
     }
 }
 
+/**
+   Checks whether the given
+   $(XREF_PACK_NAMED range,primitives,isInputRange,input range) ends with an
+   element that fulfils predicate $(D pred).
+
+   Params:
+
+   pred = Unary mandatory predicate.
+
+   doesThisEnd = The input range to check.
+
+   Returns:
+
+   $(D bool) if the first element in range $(D doesThisEnd) fulfils predicate
+   $(D pred), $(D false) otherwise.
+*/
+bool endsWith(alias pred = "a", R)(R doesThisEnd)
+    if (isInputRange!R &&
+        is(typeof(unaryFun!pred(doesThisEnd.front)) : bool))
```

Comment:
This constraint is not enough, as I found out the hard way [recently](http://stackoverflow.com/questions/33046525/why-cant-i-use-std-algorithm-count-with-a-predicate-function). This code won't compile although it should, because `among` returns a `ulong` which for some reason does not implicitly cast to bool:

```
//Error: cannot deduce type...
"asdf".startsWith!(c => c.among!('a', 'b', 'c'));
```

It'd probably be better to change it to:
`if (isInputRange!R && __traits(compiles, cast(bool)unaryFun!pred(doesThisStart.front)))`

Or something similar.